src/blocks/video.js

```diff
@@ -9,11 +9,11 @@ module.exports = Block.extend({
   // more providers at https://gist.github.com/jeffling/a9629ae28e076785a14f
   providers: {
     vimeo: {
-      regex: /(?:http[s]?:\/\/)?(?:www.)?vimeo.com\/(.+)/,
+      regex: /(?:http[s]?:\/\/)?(?:www.)?vimeo\.co(?:.+(?:\/)([^\/].*)+$)/,
```

Comment:
Yep, pop a list of the URLs it needs to match somewhere and I'll doubtless have a go.